x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/vulnerabilities.tsx

```diff
@@ -4,17 +4,236 @@
+      columnId: typeof columns[number]['id'];
+    }) => {
+      const vulnerability = data?.page[rowIndex] as VulnerabilityRecord;
+
+      if (!vulnerability) return null;
+
+      if (columnId === 'actions') {
+        return (
+          <EuiButtonIcon
+            iconType="expand"
+            aria-label="View"
+            onClick={() => {
+              alert(`Flyout id ${vulnerability.finding?.vulnerability?.id}`);
+            }}
+          />
+        );
+      }
+      if (columnId === 'vulnerability') {
+        return vulnerability.finding?.vulnerability?.id || null;
+      }
+      if (columnId === 'cvss') {
+        if (!vulnerability.finding?.vulnerability.score.base) {
+          return null;
+        }
+        return (
+          <CVSScoreBadge
+            score={vulnerability.finding.vulnerability.score.base}
+            version={vulnerability.finding?.vulnerability.score.version}
+          />
+        );
+      }
+      if (columnId === 'resource') {
+        return vulnerability.resource?.name || null;
+      }
+      if (columnId === 'severity') {
+        if (
+          !vulnerability.finding?.vulnerability.score.base ||
+          !vulnerability?.finding.vulnerability.severity
+        ) {
+          return null;
+        }
+        return (
+          <SeverityStatusBadge
+            score={vulnerability.finding.vulnerability.score.base}
+            status={vulnerability.finding.vulnerability.severity}
+          />
+        );
+      }
+      if (columnId === 'package-version') {
+        return (
+          <>
+            {vulnerability.finding?.vulnerability.package.name}{' '}
+            {vulnerability.finding?.vulnerability.package.version}
+          </>
+        );
+      }
+      if (columnId === 'fix-version') {
+        return (
+          <>
+            {vulnerability.finding?.vulnerability.package.name}{' '}
+            {vulnerability.finding?.vulnerability.package.fixed_version}
+          </>
+        );
+      }
+    };
+  }, [data?.page]);
+
+  const error = queryError || null;
+
+  if (error) {
+    return <ErrorCallout error={error as Error} />;
+  }
+  if (isLoading || !data) {
+    return <EuiLoadingSpinner />;
+  }
+
+  const columns = getVulnerabilitiesColumns();
+
   return (
     <>
-      <EuiSpacer size="l" />
-      <EuiCallOut
-        title="Placeholder for the Vulnerabilities Management integration content"
-        iconType="iInCircle"
+      <FindingsSearchBar
+        dataView={dataView}
+        setQuery={(newQuery) => {
+          setUrlQuery({ ...newQuery, pageIndex: 0 });
+        }}
+        loading={isLoading}
       />
+      <EuiSpacer size="l" />
+      {!isLoading && data.page.length === 0 ? (
+        <EuiEmptyPrompt
+          iconType="logoKibana"
+          title={
+            <h2>
+              <FormattedMessage
+                id="xpack.csp.findings.resourceFindings.noFindingsTitle"
+                defaultMessage="There are no Findings"
```

Comment:
Nit: Findings are generic for configurations or vulnerabilities. Maybe here we can be more specific with `There are no vulnerabilities`. This probably needs a second opinion from the product